### PR TITLE
Reduce code complexity

### PR DESCRIPTION
**Pull request by jmetertea**
Reduce code complexity: remove if statements, remove constant, remove new String

## Motivation and Context
Reduce code complexity

## How Has This Been Tested?
Function Tests

## Types of changes
- Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [X] My code follows the [code style][style-guide] of this project.
- [X] I have updated the documentation accordingly.

[style-guide]: https://wiki.apache.org/jmeter/CodeStyleGuidelines

### CODE REVIEW

File: src/functions/src/main/java/org/apache/jmeter/functions/DigestEncodeFunction.java [hunk 1]
```diff
@@ -103,12 +100,9 @@ public String execute(SampleResult previousResult, Sampler currentSampler) throw
...
      */
     private String uppercase(String encodedString, CompoundVariable[] values, int index) {
-        if (values.length > index) {
-            String shouldUpperCase = values[index].execute();
-            boolean shouldDoUpperCase = Boolean.TRUE.toString().equalsIgnoreCase(shouldUpperCase);
-            if (shouldDoUpperCase) {
-                return encodedString.toUpperCase();
-            }
+        String shouldUpperCase = values.length > index ? values[index].execute() : null;
+        if (Boolean.TRUE.toString().equalsIgnoreCase(shouldUpperCase)) {
```
Comment (vlsi) [inline src/functions/src/main/java/org/apache/jmeter/functions/DigestEncodeFunction.java]: Is this `Boolean.parseBoolean(shouldUpperCase)`?

Comment (jmetertea) [reply src/functions/src/main/java/org/apache/jmeter/functions/DigestEncodeFunction.java]: @vlsi yes, created a new Pull

Comment (vlsi) [reply src/functions/src/main/java/org/apache/jmeter/functions/DigestEncodeFunction.java]: Why do you create a new one? Would you please just update the current PR?

**Comment by pmouawad** [comment]
This duplicates #516 
Closing this one

**Pull request closed without merging**